src/python/era_5g_object_detection_standalone/era_5g_object_detection_standalone/interface.py

```diff
@@ -121,14 +125,85 @@ def post_image():
     for file in request.files.to_dict(flat=False)['files']:
         # print(file)
         nparr = np.fromstring(file.read(), np.uint8)
-        # decode image
-        img = cv2.imdecode(nparr, cv2.IMREAD_COLOR)
 
         # store the image to the appropriate task
-        task.store_image({"sid": sid, "websocket_id": task.websocket_id, "timestamp": timestamps[index]}, img)
+        # the image is not decoded here to make the callback as fast as possible
+        task.store_image({"sid": sid, 
+                          "websocket_id": task.websocket_id, 
+                          "timestamp": timestamps[index], 
+                          "decoded": False}, 
+                         nparr)
         index += 1
     return Response(status=204)
 
+@socketio.on('connect', namespace='/data')
+def connect_ws(auth):
+    """_summary_
+    Creates a websocket connection to the client for passing the data.
+
+    Raises:
+        ConnectionRefusedError: Raised when attempt for connection were made
+            without registering first.
+    """
+
+    if 'registered' not in session:
+        raise ConnectionRefusedError('Need to call /register first.')
+    
+    print(f"Connected data. Session id: {session.sid}, ws_sid: {request.sid}")
+    
+    flask_socketio.send("you are connected", namespace='/data', to=request.sid)
+
+@socketio.on('image', namespace='/data')
+def image_from_client(data: dict):
+    """
+    Allows to receive jpg-encoded image using the websocket transport
+
+    Args:
+        data (dict): A base64 encoded image frame and (optionally) related timestamp in format:
+            {'frame': 'base64data', 'timestamp': 'int'}
+
+    Raises:
+        ConnectionRefusedError: Raised when attempt for connection were made
+            without registering first or frame was not passed in correct format.
+    """
+    logger.debug("A frame recieved using ws")
+    if 'registered' not in session:
+        raise ConnectionRefusedError('Need to call /register first.')
+    if 'frame' not in data:
+        raise ConnectionRefusedError('Data does not contain frame')
+    if 'timestamp' in data:
+        timestamp = data['timestamp']
+    else:
+        logger.debug("Timestamp not set, setting default value")
+        timestamp = 0
+
+    task = tasks[session.sid]
+    try:
+        frame = base64.b64decode(data["frame"])
+        task.store_image({"sid": session.sid, 
+                          "websocket_id": task.websocket_id, 
+                          "timestamp": timestamp, 
+                          "decoded": False}, 
+                         np.frombuffer(frame, dtype=np.uint8))
+    except (ValueError, binascii.Error) as error:
+        logger.error(f"Failed to decode frame data: {error}")
+
```

Comment:
How does the function return an error or response to the client?

Reply:
We can send response and an appropriate callback to the client.